apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainApplication.kt

```diff
@@ -0,0 +1,112 @@
+
+import expo.modules.ApplicationLifecycleDispatcher
+import expo.modules.ReactNativeHostWrapper
+
+import java.util.Arrays
+//MainApplication.java
+import com.adobe.marketing.mobile.AdobeCallback
+import com.adobe.marketing.mobile.Assurance
+import com.adobe.marketing.mobile.Edge
+import com.adobe.marketing.mobile.Extension
+import com.adobe.marketing.mobile.Lifecycle
+import com.adobe.marketing.mobile.LoggingMode
+import com.adobe.marketing.mobile.Messaging
+import com.adobe.marketing.mobile.MobileCore
+import com.adobe.marketing.mobile.Places
+import com.adobe.marketing.mobile.Signal
+import com.adobe.marketing.mobile.Target
+import com.adobe.marketing.mobile.UserProfile
+import com.adobe.marketing.mobile.edge.bridge.EdgeBridge
+import com.adobe.marketing.mobile.edge.consent.Consent
+import com.adobe.marketing.mobile.edge.identity.Identity
+import com.adobe.marketing.mobile.optimize.Optimize
+import com.anonymous.AEPSampleAppNewArchEnabled.BuildConfig
+
+
+class MainApplication : Application(), ReactApplication {
+
+  override val reactNativeHost: ReactNativeHost = ReactNativeHostWrapper(
+        this,
+        object : DefaultReactNativeHost(this) {
+          override fun getPackages(): List<ReactPackage> {
+            // Packages that cannot be autolinked yet can be added manually here, for example:
+            // packages.add(new MyReactNativePackage());
+            return PackageList(this).packages
+          }
+
+          override fun getJSMainModuleName(): String = ".expo/.virtual-metro-entry"
+
+          override fun getUseDeveloperSupport(): Boolean = BuildConfig.DEBUG
+
+          override val isNewArchEnabled: Boolean = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
+          override val isHermesEnabled: Boolean = BuildConfig.IS_HERMES_ENABLED
+      }
+  )
+
+  override val reactHost: ReactHost
+    get() = ReactNativeHostWrapper.createReactHost(applicationContext, reactNativeHost)
+
+  override fun onCreate() {
+    super.onCreate()
+
+      MobileCore.setApplication(this);
+    MobileCore.setLogLevel(LoggingMode.DEBUG)
+    MobileCore.configureWithAppID("example")
+    val extensions: List<Class<out Extension?>> = Arrays.asList(
+      Lifecycle.EXTENSION,
+      Signal.EXTENSION,
+      Edge.EXTENSION,
+      Identity.EXTENSION,
+      Consent.EXTENSION,
+      EdgeBridge.EXTENSION,
+      Messaging.EXTENSION,
+      UserProfile.EXTENSION,
+      Assurance.EXTENSION,
+      Places.EXTENSION,
+      Target.EXTENSION,
+      Optimize.EXTENSION,
+      com.adobe.marketing.mobile.Identity.EXTENSION
+    )
+    MobileCore.registerExtensions(extensions,
+      AdobeCallback { o: Any? ->
+        MobileCore.lifecycleStart(
+          null
+        )
+      })
+    SoLoader.init(this, false)
+    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
+      // If you opted-in for the New Architecture, we load the native entry point for this app.
+      load()
+    }
+    ApplicationLifecycleDispatcher.onApplicationCreate(this)
+  }
```

Comment:
Lifecycle is not implemented correctly. 
https://developer.adobe.com/client-sdks/home/base/mobile-core/lifecycle/#register-lifecycle-with-mobile-core-and-add-appropriate-startpause-calls
As mentioned in the doc, we should add below code to the `MainApplication` class:
```java
    @Override
    public void onResume() {
        MobileCore.setApplication(getApplication());
        MobileCore.lifecycleStart(null);
    }
```
and 
```java
    @Override
    public void onPause() {
        MobileCore.lifecyclePause();
    }
```

Reply:
`    ApplicationLifecycleDispatcher.onApplicationCreate(this)

    // Register activity lifecycle callbacks to handle onResume and onPause
    registerActivityLifecycleCallbacks(object : ActivityLifecycleCallbacks {
      override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}
      override fun onActivityStarted(activity: Activity) {}
      override fun onActivityResumed(activity: Activity) {
        MobileCore.setApplication(this@MainApplication)
        MobileCore.lifecycleStart(null)
      }
      override fun onActivityPaused(activity: Activity) {
        MobileCore.lifecyclePause()
      }
      override fun onActivityStopped(activity: Activity) {}
      override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
      override fun onActivityDestroyed(activity: Activity) {}
    })
  }
`

have to use `registerActivityLifecycleCallbacks` because onResume() and onPause() methods are not part of the Application class, which MainApplication extends. These methods are actually part of the Activity class.

Reply:
Oh, my bad. Those methods are part of the Activity class. The suggested approach is to override these methods within the 'MainActivity", as mentioned in the public docs: https://developer.adobe.com/client-sdks/home/base/mobile-core/lifecycle/android/#implementing-global-lifecycle-callbacks
I'm not 100% sure about the behavior of the `registerActivityLifecycleCallbacks` class; I assume it's bundled into each Activity. But, the mobile SDK only monitors the App lifecycle. So, I still recommend referring to the SDK documentation mentioned above to add a MainActivity class and override the `onResume` and `onPause` methods.